src/main/java/uk/nhs/hee/tis/trainee/credentials/model/ModificationMetadata.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2023 Crown Copyright (Health Education England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.trainee.credentials.model;
+
+import java.time.Instant;
+import org.springframework.data.annotation.Id;
+import uk.nhs.hee.tis.trainee.credentials.dto.CredentialType;
+
+/**
+ * Metadata of modified data related to credentials.
+ */
+public record ModificationMetadata(@Id ModificationMetadataId id, Instant lastModifiedDate) {
+
+  /**
+   * Create a metadata object for data modification.
+   *
+   * @param tisId            The TIS ID of the modified data.
+   * @param credentialType   The credential type related to the data object.
+   * @param lastModifiedDate The instant of the modification.
```

Comment:
Again, this might be jumping the gun a bit, but wouldn't it be useful to track what sort of modification it is: delete or update?

Updates are going to be a bit more interesting, since I think we'll need to determine if its a 'meaningful' update, i.e. it affects once of the fields included in the credential, otherwise we can probably ignore the event.